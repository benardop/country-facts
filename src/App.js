import './App.css';
import { Route, Routes } from 'react-router-dom';
import Home from './components/Home/Home';
import Header from './components/Header/Header';
import Details from './components/Details/Details';

function App() {
  return (
    <div className="App">
      <Header />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/details" element={<Details />} />
      </Routes>

    </div>
  );
}

export default App;
