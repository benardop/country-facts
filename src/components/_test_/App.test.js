import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import App from '../../App';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if Item component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the App container is there', async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('app-container');
  expect(container).toBeInTheDocument();
});
