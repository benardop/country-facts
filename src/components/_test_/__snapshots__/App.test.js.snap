// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Item component has changed 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="left arrow"
        src="leftArrow.svg"
      />
    </a>
    <span>
      Country facts
    </span>
    <div>
      <img
        alt="microphone"
        src="mic.svg"
      />
      <img
        alt="three dots"
        src="threeDots.svg"
      />
    </div>
  </div>
  <div>
    <div
      className="dropDown"
    >
      <select
        name="drop-down"
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="Asia"
        >
          Asia
        </option>
        <option
          value="Africa"
        >
          Africa
        </option>
        <option
          value="Americas"
        >
          America
        </option>
        <option
          value="Europe"
        >
          Europe
        </option>
        <option
          value="Oceania"
        >
          Oceania
        </option>
      </select>
    </div>
    <div
      className="container"
    />
  </div>
</div>
`;
